### PR TITLE
Many to many

### PR DESCRIPTION
**Pull request by rochedarran**
Add a many to many relationship.
Entities involved: Allergy, and Allergies for a Patient.

@Audit test done for Allergy and Patient but could't add any for new entity PatientAllergy.

Patients DB purged has been changed to be done every first of January at 6 am.

### CODE REVIEW

**Comment by quimnoto** [comment]
I have check the conflict it was one import of LoaclDate and it has been solved

File: src/main/java/com/systelab/seed/controller/PatientController.java [hunk 1]
```diff
@@ -32,40 +47,70 @@ public PatientController(PatientService patientService) {
...
     @PutMapping("patients/{uid}")
     public ResponseEntity<Patient> updatePatient(@PathVariable("uid") UUID id, @RequestBody @ApiParam(value = "Patient", required = true) @Valid Patient p) {
         Patient patient = this.patientService.updatePatient(id, p);
         URI selfLink = URI.create(ServletUriComponentsBuilder.fromCurrentRequest().toUriString());
         return ResponseEntity.created(selfLink).body(patient);
     }
 
+    @ApiOperation(value = "Add an alergy to a Patient", authorizations = { @Authorization(value = "Bearer") })
+    @PutMapping("patients/{patientUid}/allergies/{allergyUid}")
+    public ResponseEntity<Patient> addAlergyToPatient(@PathVariable("patientUid") UUID patientId, @PathVariable("allergyUid") UUID allergyId,
```
Comment (alfonsserra) [inline src/main/java/com/systelab/seed/controller/PatientController.java]: In the specification, two operations are specified, by POST and by PUT. It is important to stick to the specification in order to work with the frontend.

File: src/main/java/com/systelab/seed/service/PatientService.java [hunk 2]
```diff
@@ -41,18 +53,66 @@ public Patient createPatient(Patient p) {
...
+
+        patient.getAllergies().add(patientAllergy);
+
+        return this.patientRepository.save(patient);
+    }
+
+    public Set<Allergy> getAllergiesFromPatient(UUID patientId) {
+        Set<Allergy> allergies = new HashSet<>();
+        Patient patient = this.getPatient(patientId);
+        patient.getAllergies().stream().forEach(pa -> allergies.add(pa.getAllergy()));
```
[the remark this answers is not in the document]
Comment (rochedarran) [reply src/main/java/com/systelab/seed/service/PatientService.java]: :-o  solved :-) by the way, we don't need this method, as we have change a bit the implementation after a conversation between @alfonsserra and @quimnoto

File: src/main/java/com/systelab/seed/controller/PatientController.java (hunk 1 above)
Comment (rochedarran) [reply src/main/java/com/systelab/seed/controller/PatientController.java]: Operations added as mentioned in the specification

**Pull request merged**